title_belt_nhl/schedule.py

```diff
@@ -11,33 +11,63 @@
 SCHEDULE_FILE = Path(__file__).parent / "static" / "schedule_2024_2025.csv"
 
 
-class TitleBelt:
-    team: str
-    belt_holder: str
-
-    def __init__(self, team, belt_holder):
-        self.team = team
-        self.belt_holder = belt_holder
-
-
 class Match:
     home: str
     away: str
-    date: int
+    serial_date: int
+    date_obj: date
+    belt_holder: str = None
+    home_last: str = None
+    away_last: str = None
 
-    def __init__(self, home, away, date):
+    def __init__(self, home, away, serial_date=None, date_obj=None):
         self.home = home
         self.away = away
-        self.date = date
+        self.serial_date = serial_date
+        self.date_obj = date_obj or ExcelDate(serial_date=serial_date).date_obj
 
     def __str__(self):
         return f"[{self.home} vs {self.away}]"
 
 
+def traverse_matches_backwards(
+    matches: list[list[Match]] | None = None, match: Match | None = None
+):
+    """Traverse a tree/graph of matches backwards to find the path to the top.
+
+    Parameters:
+    - matches: list[list[Match]] (optional)
+      - each index of matches represents a depth into the tree/graph of matches
+      - index 0 and last index should each contain a single match
+      - index 0 represents the upcoming match for the current belt holder
+      - last index represents the first match in the graph where the current team
+        can play for the belt
+    - match: Match (optional)
+      - basically same as above, but only pass the last match that we can work up from
+
+    ASSUMPTIONS:
+    - each match specifies either home_last or away_last, matching with the
+      match's belt_holder (home_last if belt_holder is home, or vice versa)
+    """
+    if not matches and not match:
+        raise ValueError("Either matches or match must be provided!")
+    path_matches = []
+    cur_match = match or matches[-1][0]
+    while cur_match:
+        path_matches.insert(0, cur_match)
+
+        last_match = cur_match.away_last or cur_match.home_last
```

Comment:
I assume this will just take `.away_last` if both `.away_last` and `.home_last` are present?  Does there need to be a flag that helps determine which one will lead back to the original belt holder?

Reply:
Yea with how it works now this should work, but it probably makes more sense to be a bit more explicit with something like:

```py
last_match = cur_match.away_last if cur_match.belt_holder == cur_match.away else cur_match.home_last
```